### PR TITLE
Dynamic radio options from checkbox answers

### PR DESCRIPTION
**Pull request by LJBabbage**
### What is the context of this PR?
This PR introduces the ability to for radio answers to use checkbox answers as their options

### How to review 
Using test_dynamic_radio_options_from_checkbox.json make sure the answers you select in the checkboxes appears as 
options in the radio question. The first radio question has an additional static answer as the 2nd radio only has dynamic answers.

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

File: tests/app/forms/field_handlers/test_dropdown_handler.py [hunk 1]
```diff
@@ -102,3 +104,17 @@ class TestForm(Form):
...
+    value_source_resolver, rule_evaluator
+):
+    handler = DropdownHandler(
+        dynamic_radio_options_no_static_options(),
+        value_source_resolver,
+        rule_evaluator,
+        error_messages,
+    )
+    with pytest.raises(InvalidSchemaConfigurationException) as context:
+        handler.choices  # pylint: disable=pointless-statement
```
Comment (MebinAbraham) [inline tests/app/forms/field_handlers/test_dropdown_handler.py:119]: Just need asserting.
```suggestion
        assert handler.choices
```

File: tests/app/forms/field_handlers/test_select_handler.py [hunk 2]
```diff
@@ -62,3 +64,17 @@ class TestForm(Form):
...
+    value_source_resolver, rule_evaluator
+):
+    handler = SelectHandler(
+        dynamic_radio_options_no_static_options(),
+        value_source_resolver,
+        rule_evaluator,
+        error_messages,
+    )
+    with pytest.raises(InvalidSchemaConfigurationException) as context:
+        handler.choices  # pylint: disable=pointless-statement
```
Comment (MebinAbraham) [inline tests/app/forms/field_handlers/test_select_handler.py:79]: Just need asserting
```suggestion
        assert handler.choices
```

File: app/questionnaire/dynamic_answer_options.py [hunk 3]
```diff
@@ -1,24 +1,38 @@
...
-        if "source" in values:  # pylint: disable=no-else-raise
-            # :TODO: Implement value sources support
-            # resolved_values = ...
-            raise NotImplementedError  # pragma: no cover
+        resolved_values: Union[
+            ValueSourceEscapedTypes, ValueSourceTypes, RuleEvaluatorTypes
+        ]
+
+        if "source" in values:
+            if not values["source"] == "answers":
```
Comment (berroar) [inline app/questionnaire/dynamic_answer_options.py:26]: Can be simplified
```suggestion
            if values["source"] != "answers":
```

File: tests/functional/spec/features/dynamic_answer_options/radio_options_from_checkbox_answers.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      $(InjurySustainedPage.submit()).click();
+
+      expect(browser.getUrl()).to.contain(MostSeriousInjuryPage.pageName);
+      expect($(MostSeriousInjuryPage.answerLabelByIndex(0)).getText()).to.contain("Head");
+      expect($(MostSeriousInjuryPage.answerLabelByIndex(1)).getText()).to.contain("Body");
+      expect($(MostSeriousInjuryPage.answerLabelByIndex(2)).getText()).to.contain("They were of equal severity (static option)");
+      expect($(MostSeriousInjuryPage.answerLabelByIndex(3)).isExisting()).to.be.false;
+    });
+
+    it("When the respondent answers the radio question and submits, Then the radio question should show only the answers from the first checkbox as options", () => {
```
Comment (MebinAbraham) [inline tests/functional/spec/features/dynamic_answer_options/radio_options_from_checkbox_answers.js:21]: Might be clearer if it said `Then the next radio question..`, found it confusing as you are on a radio, so made me this it was somehow still the same radio.

File: app/questionnaire/questionnaire_schema.py [hunk 5]
```diff
@@ -10,6 +10,16 @@
...
 from app.questionnaire.rules.operator import OPERATION_MAPPING
 
+
+class InvalidSchemaConfigurationException(Exception):
+    def __init__(self, value: Union[str, int]) -> None:
+        super().__init__()
+        self.value = value
+
+    def __str__(self) -> str:
+        return str(self.value)
```
Comment (MebinAbraham) [inline app/questionnaire/questionnaire_schema.py:20]: We should need this override, as inheriting from `Expection` should be enough. 

I think we should move this class below the constants defined below.

**Review by MebinAbraham — COMMENTED** [review]
Looks good, minor issues.

**Review by parimooa — COMMENTED** [review]
LGTM, Working as expected

**Comment by jamesnunn** [comment]
LGTM, obviously the backwards journey not being implemented yet makes changing answers behaviour surprising but I think that's just how it is until that's addressed.

**Review by jamesnunn — APPROVED** [review]

File: scripts/run_validator.sh [hunk 6]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
-tag=latest
+tag=add-dynamic-options-identifier-check
```
Comment (MebinAbraham) [inline scripts/run_validator.sh:2]: Reminder to revert before merge.
```suggestion
tag=latest
```

**Review by MebinAbraham — APPROVED** [review]

**Review by berroar — APPROVED** [review]

**Pull request merged**